crates/network/src/ethereum/builder.rs

```diff
@@ -214,6 +222,10 @@ fn build_2930(request: TransactionRequest) -> Result<TxEip2930, TransactionBuild
     })
 }
 
+const fn will_build_4844(request: &TransactionRequest) -> bool {
+    request.sidecar.is_some()
+}
+
 /// Build an EIP-4844 transaction.
 fn build_4844(request: TransactionRequest) -> Result<TxEip4844, TransactionBuilderError> {
```

Comment:
related to other comment:
I don't like that these functions will return only 1 missing key per invocation. makes it annoying to debug fill strategies, etc. It'd be nice if we could use `can_build_4844` to find all missing keys in one invocation

Reply:
If I understand correctly, `TransactionBuilderError::MissingKey(&'static str)` should be changed to   `TransactionBuilderError::MissingKeys(Vec<&'static str>)`, right ?

Reply:
yes, that is how i would do it